src/app/(spaces)/homebase/PrivateSpace.tsx

```
@@ -85,31 +87,36 @@ function PrivateSpace({ tabName, castHash }: { tabName: string; castHash?: strin
-      await loadTab(tabName);
-    }
-
-    // After the current tab is loaded, preload other tabs in the background
-    void loadRemainingTabs();
+  // Load tab config without blocking the initial render
+  function loadTabConfigAsync() {
+    startTransition(() => {
+      (async () => {
+        await loadTabNames();
+        // Re-read current ordering to avoid stale state
+        const freshTabOrdering = useAppStore(state => state.homebase.tabOrdering);
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)